src/replay/service.zig

```diff
@@ -830,6 +832,18 @@ fn bypassConsensus(state: *ReplayState, results: []const ReplayResult) !bool {
         state.logger.info().logf("rooting slot with SlotTree.reRoot: {}", .{new_root});
         slot_tracker.root = new_root;
         slot_tracker.pruneNonRooted(state.allocator);
+        if (state.account_store == .accounts_db) {
+            // TODO: get this value from the right place
+            const lamports_per_signature = 5000;
+
+            try sig.accounts_db.manager.onSlotRooted(
```

Comment:
Could you make this an interface method? Replay does not depend on AccountsDB, and I'd like to keep it that way.

Reply:
Another approach to rooting might actually fix this pre-existing bug. If you delete every account between the last rooted slot and the currently rooted slot, then it won't have the bug described in that ticket. This makes an assumption that the caller to onSlotRooted will actually call that function for every single slot that needs to be rooted, and if slots are skipped, then they truly need to be discarded.

Reply:
I don't think we need to block the merge of this PR because of this pre-existing bug. But it would be nice to get this fixed soon.